File: src/main/java/net/dontdrinkandroot/gitki/wicket/component/item/UserListPageItem.java
package net.dontdrinkandroot.gitki.wicket.component.item;

import net.dontdrinkandroot.gitki.model.Role;
import net.dontdrinkandroot.gitki.wicket.page.admin.UserListPage;
import net.dontdrinkandroot.gitki.wicket.security.Render;
import net.dontdrinkandroot.wicket.bootstrap.component.item.BookmarkablePageLinkItem;
import org.apache.wicket.Page;
import org.apache.wicket.model.StringResourceModel;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
@Render(Role.ADMIN)
public class UserListPageItem extends BookmarkablePageLinkItem
{
    public <C extends Page> UserListPageItem(String id)
    {
        super(id, new StringResourceModel("gitki.users"), UserListPage.class);
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/component/item/UserListPageItem.kt
package net.dontdrinkandroot.gitki.wicket.component.item

import net.dontdrinkandroot.gitki.model.Role
import net.dontdrinkandroot.gitki.wicket.page.admin.UserListPage
import net.dontdrinkandroot.gitki.wicket.security.Render
import net.dontdrinkandroot.wicket.bootstrap.component.item.BookmarkablePageLinkItem
import org.apache.wicket.model.StringResourceModel

@Render(Role.ADMIN)
class UserListPageItem(id: String) :
    BookmarkablePageLinkItem<Any>(
        id,
        label = StringResourceModel("gitki.users"),
        pageClass = UserListPage::class.java
    )